src/main/java/org/hyperledger/fabric/sdk/BlockDeserializer.java

```diff
@@ -81,8 +79,12 @@ EnvelopeDeserializer getData(int index) throws InvalidProtocolBufferException {
     }
 
     byte[] getTransActionsMetaData() {
-
-        return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        List<ByteString> metadataList = block.getMetadata().getMetadataList();
+        if (metadataList != null && metadataList.size() > Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE) {
+            return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        } else {
+            return metadataList.get(0).toByteArray();
```

Comment:
You are adding code to do bounds checking on the metadata but then failing to check whether the zero index element exists before using it. Regardless, this is absolutely the wrong thing to return. You must return `getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray()` or throw an exception.